verilog/tools/kythe/scope_resolver.h

```diff
@@ -168,7 +175,28 @@ class FlattenedScopeResolver {
   void MapSignatureToScopeOfSignature(const Signature& signature,
                                       const Signature& other_signature);
 
- private:
+  ScopeContext& GetScopeContext() { return scope_context_; }
```

Comment:
This is changed during traversal so it has to bee mutable.

Reply:
From a class-design perspective, try to avoid this if possible because this may invite the caller to violate class-invariant properties (like relationships between members).  It takes more analysis to conclude whether or not this is possible.  If the number of mutating operations on `ScopeContext` is sufficiently small, consider expose only those mutable operations.

In this case, I would name this `GetMutableScopeContext()` to convey intent more clearly where it is called.